### ISSUE TITLE
Provide default record values for Lint API

### ISSUE THREAD

**Issue by dungpa** [fsprojects/FSharpLint#105]
We currently use:

```fsharp
Lint.lintParsedFile 
                    { FinishEarly = None
                      Configuration = None //Configuration.defaultConfiguration
                      ReceivedWarning = None }
                    { Ast = ast
                      Source = source
                      TypeCheckResults = checkResults.GetCheckResults() }
```

It would be great if `OptionalLintParameters` record type has a `Default` property (providing minimally useful config) so that we could just use `Lint.lintParsedFile OptionalLintParameters.Default ...`.

Any customization can be easily done with `{ OptionalLintParameters.Default with ... }`.

The same thing could be applied for `FSharpLint.Framework.Configuration.Configuration` record type. At a first glance, it's hard for clients to figure out what should be passed in.

**Comment by duckmatt** [comment]
Thanks that's a good suggestion. I've added a `Default` property. 

`Configuration = None` will use `Configuration.defaultConfiguration` when linting (I've just added this to the xml doc for the record) so there is no need to pass in a config unless it's custom (custom configs can be loaded using the functions in `FSharpLint.Application.ConfigurationManagement` in `lint.fsi`)

**Comment by dungpa** [comment]
Great, thanks.
